### PR TITLE
feat!: robust "firstboot" system

### PR DESCRIPTION
**Pull request by Arcitec**
Everything is deeply tested in a Virtual Machine and works beautifully. The new "firstboot" system is protected (immutable), fixes the issue of legacy leftover scripts, and automatically upgrades all legacy systems to the new link system.

The "justfile" was clashing with "ublue-os/main" in an undesirable way, and it was decided to prefix each justfile with the name of the image/fork, so that has also been fixed here.

People also thought I should include the design flowchart for the new "firstboot" system, so that's also included as a highly compressed little PNG file. Start by [looking at that image](https://github.com/ublue-os/startingpoint/pull/74/commits/44148eb0c21bb6d3996d657436fafad29360333e) and everything will be clear.

This concludes the standardization procedure! After this is merged, we're flexible and fully up to speed with uBlue's direction!

### CODE REVIEW

**Review by castrojo — APPROVED** [review]
Overall this is awesome! +1 from me but let's give @EinoHR a chance to weigh in since there's a timezone difference that we need to account for. 

Thanks so much for this!

**Comment by Arcitec** [comment]
I went through everything one last time with a fine-toothed comb to look for anything that could be improved grammatically or anything else. The last two commits are new, but everything else that "appears to be new" is actually just squashed/slightly reworded files that Jorge has already checked. Phew. It's been a lot of work. Enjoy! :)

File: README.md [hunk 1]
```diff
@@ -21,10 +21,10 @@ The easiest way to start customizing is by looking at and modifying `recipe.yml`
...
 
 > **Note**
-> The configuration files you put in the `etc` directory are actually added to `/usr/etc/` where they get applied to your local `/etc/` when rebasing to or updating the image. If a config file in `/etc/` has been changed, the changes won't be overridden, but the new version will be available in `/usr/etc/`. Run `sudo ostree admin config-diff` to see the difference between `/etc/` and `/usr/etc/` (`man ostree-admin-config-diff` for further documentation).
+> The configuration files you put in the `usr/etc/` directory are actually added to `/usr/etc/` on the final system, where they will be automatically applied to your local `/etc/` when rebasing to or updating the image. If a config file in `/etc/` has been changed by the user, OSTree will attempt to automatically merge the user's changes into the installed file, but if that fails, the new version will only be available in `/usr/etc/`. Run `sudo ostree admin config-diff` to see the difference between `/etc/` and `/usr/etc/` (`man ostree-admin-config-diff` for further documentation).
```
Comment (EinoHR) [inline README.md:27]: This note could be slightly rephrased, as it should be obvious that files in `usr/etc/` go to `/usr/etc/`.

File: README.md [hunk 2]
```diff
@@ -97,9 +97,17 @@ The `latest` tag will automatically point to the latest build. That build will s
...
-The `just` task runner is included in `main` for further customization after first boot.
+The `just` task runner is included in `ublue-os/main`-derived images, and we have provided several template commands which help you perform further customization after first boot.
 
-You can copy the justfile from `/usr/share/ublue-os/just/justfile` to `~/.justfile` to get started. When `just` supports [include directives](https://just.systems/man/en/chapter_52.html), you will instead be able to simply include that path into your own justfile, without having to copy it manually.
+You can merge our template justfiles into your own local configuration. When `just` supports [include directives](https://just.systems/man/en/chapter_52.html), you will instead be able to simply include these paths into your own justfile, without having to copy anything manually.
+
+Run the following commands when you're logged into the operating system, to merge uBlue's provided configurations into your own user config. (The "touch" command is only necessary on certain shells which won't let you merge into non-existent files.)
+
+```sh
+touch ~/.justfile
```
Comment (EinoHR) [inline README.md:107]: Nice!

File: usr/share/ublue-os/firstboot/yafti.yml [hunk 3]
```diff
@@ -67,24 +74,25 @@ screens:
...
-            - MangoHUD: org.freedesktop.Platform.VulkanLayer.MangoHud//22.08
             - Steam: com.valvesoftware.Steam
-            - Gamescope for Steam: com.valvesoftware.Steam.Utility.gamescope
+            - Gamescope (Utility): com.valvesoftware.Steam.Utility.gamescope
+            - MangoHUD (Utility): org.freedesktop.Platform.VulkanLayer.MangoHud//22.08
+            - SteamTinkerLaunch (Utility): com.valvesoftware.Steam.Utility.steamtinkerlaunch
             - Proton Updater for Steam: net.davidotek.pupgui2
         Office:
-          description: Bow to Capitalism
+          description: Boost your productivity.
```
Comment (EinoHR) [inline usr/share/ublue-os/firstboot/yafti.yml:89]: Revert this change NOW :gun:. Oh shoot, that's a water pistol... Well, wouldn't you agree that "Bow to Capitalism" and "Boost your productivity" mean the same thing with a difference in personality...

File: recipe.yml [hunk 4]
```diff
@@ -1,11 +1,11 @@
-# Warning: only edit this file inside the images git repository.
-# If you are looking at this file in your root directory on your computer, editing it does nothing.
+# Warning: Only edit this file inside the uBlue image's git repository.
```
Comment (EinoHR) [inline recipe.yml:1]: This warning is useless, as now the recipe resides in an immutable directory.

**Comment by EinoHR** [comment]
Thanks @Arcitec! I was about to start fixing my complaints, but you don't have *allow maintainers to push commits* on.

File: README.md (hunk 1 above)
Comment (Arcitec) [reply README.md:27]: @EinoHR Yeah, I followed the flow of the original note but agree that I also felt it was a bit redundant. Fixed. :)

File: README.md (hunk 2 above)
Comment (Arcitec) [reply README.md:107]: @EinoHR Yeah, I also force-pushed a change which changes the suffix to `.just` instead, to follow upstream "just" project's standards. We'll soon be able to just include those files beautifully as modules without any of these merging tricks. :)

File: usr/share/ublue-os/firstboot/yafti.yml (hunk 3 above)
Comment (Arcitec) [reply usr/share/ublue-os/firstboot/yafti.yml:89]: @EinoHR Yeah it was a bit funny, but we should be neutral and set a good example. :)

File: recipe.yml (hunk 4 above)
Comment (Arcitec) [reply recipe.yml:1]: @EinoHR Very good point, removing the recipe.yml edit warning since it's no longer in a mutable directory. :)

**Comment by Arcitec** [comment]
> Thanks @Arcitec! I was about to start fixing my complaints, but you don't have _allow maintainers to push commits_ on.
> 
> Overall the code and changes look fantastic! I'm just ~too~ keen on the documentation side o' things

Hey, great to hear from you! I don't mind the strict feedback at all! You and I both care a lot about clear documentation, and there's no such thing as "caring too much about documentation". :) The users will be happier in the end.

I've addressed the comments, and also did a rebase to fix the `.justfile -> .just` suffix of the modular just file, after discovering that it's how upstream "just" projects officially name things.

So: Two new commits to address the feedback, and an edit of the justfile-renaming commit to use the correct new suffix. That's it! :)

**Comment by Arcitec** [comment]
After discovering the bug/quirk in yafti where "duplicate Flatpak names break the installer", I've also cleaned up the duplicate package names now. Spoke to Marco about it on Discord and he'll be marking duplicate yafti yml packages as a validation error, which means that yafti won't even run when there are duplicates in the specification.

There's a ticket here to track that work: https://github.com/ublue-os/yafti/issues/99

Either way, startingpoint's yafti yml is now de-duplicated and will work properly in the future when that change is live.

**Comment by EinoHR** [comment]
LGTM.

PS: I'd prefer "why" documentation inline instead of in the commit message, not everyone will be `git blame`-ing their way through the repo to see _why this line is the way it is_.

[see 1eaf70](https://github.com/ublue-os/startingpoint/pull/74/commits/1eaf70e8464aada4929d46189274bb42e5a0aa12)
> PS: The strange "run" condition is required since yafti demands that the return code of a command should be "0" to show a conditional screen, but "grep" by itself returns "1" when nothing matches. Negation operators such as `!` don't work in the run-conditions either, so this workaround does the job perfectly instead. It robustly checks for the exact number "0", meaning that the Flathub remote wasn't found.

**Comment by EinoHR** [comment]
Merging either way now.

**Comment by Arcitec** [comment]
@EinoHR Yeah, but the comment was in the commit since it was mainly to help you during the review. I was planning to make a ticket @ yafti about the issue instead. Here's the issue tracker entry for it:

https://github.com/ublue-os/yafti/issues/100

If something like that is added, we could clean that line up.

**Comment by Arcitec** [comment]
I stumbled on bsherman's repo and saw that he had also gotten rid of the "Bow to Capitalism" line. :smile: 

https://github.com/bsherman/ublue-custom/commit/a07e18a4db848ca9b924a10e3e27cb8d35a29559#diff-79009b18dec43001bbeffb07bdb4b403e07604741cc11578f40400ad25b91628L85

Capitalism wins... this battle. :1st_place_medal:

**Comment by xynydev** [comment]
> Capitalism wins... this battle. 🥇

_noooo_
